### PR TITLE
WIP: Draft ggplot2 2.3.0 upcoming announcement post.

### PR DESCRIPTION
**Pull request by batpigandme**
* Add images for ggplot2 post

### CODE REVIEW

**Comment by batpigandme** [comment]
I got really nervous about this being too long…so it might overstate what I actually cover.

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+In addition to highlighting a few features and improvements, we also want to 
+share a bit about our release-preparation process for ggplot2, which has over 
+1,700 reverse dependencies.
+
+## Breaking changes
+
+### Our process
+
+Regression testing, visual and otherwise, allows us to keep track of changes
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: The first "you" here was a bit hard to parse for me. Maybe make it "we"? Or otherwise change the lead-in.

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+At this point, we begin to manually examine the failures. "Newly broken" issues
+are extracted from the problems report and put into a Google sheet where (with
+the great help of Thomas Lin Pedersen, Claus Wilke, and Kara Woo) we identified
+clusters of errors to tackle. For example, 97 packages returned a warning due
+to a namespace clash between `dplyr::vars()` and `ggplot2::vars()`. That’s
+something we clearly need to fix, so we open an [issue](https://github.com/
+tidyverse/ggplot2/issues/2585) to track progress.
+
+Where reverse-dependency errors result from a deliberate design change, we now
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: now have -> collect

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * We decided that it’s worth breaking a small amount of existing code in the 
+ interests of improving future code. For example, this version of ggplot2 adds 
+ support for [tidy evaluation](https://github.com/tidyverse/ggplot2/blob/master/NEWS.md#tidy-evaluation) 
+ making ggplot2 more compatible with the rest of the tidyverse. This breaks a 
+ small number of packages that use ggplot2 but should affect very little data 
+ analysis code, so we decided that it was worth it.  
+
+### What to do if you run into a breaking change
+
+The maintainers of reverse dependencies that are likely to encounter a breaking
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: I think you should remove the bit about revdeps here - this is orthogonal to what a user should do if they see a problem.

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+before filing a new one to avoid duplication.
+
+You can also ask for help in [community.rstudio.com](https://community.rstudio.com/).
+
+## New features
+
+### Tidy evaluation
+
+You can now use quasiquotation in `aes()`, `facet_wrap()`, and `facet_grid()`. 
+The formula interface for faceting is not going away, but this is meant to
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Remove 2nd sentence and replace with: "To support quasiquotation in facetting we've added a new helper that works similar to `aes()`: `vars()`, short for variables and instead of `facet_grid(x + y ~  a + b)` you can now write `facet_grid(vars(x, y), vars(a, b))`. The formula interface won't go away; but the new `vars()` interface should be much easier to program with.

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+### Tidy evaluation
+
+You can now use quasiquotation in `aes()`, `facet_wrap()`, and `facet_grid()`. 
+The formula interface for faceting is not going away, but this is meant to 
+make it easier to create functions around `facet_wrap()` and `facet_grid()`, 
+including label titles for facets with variable names and values.
+
+```{r label_both}
+p <- ggplot(mtcars, aes(wt, mpg)) + geom_point()
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Replace with something like

```R
x_var <- quo(wt)
y_var <- quo(mpg)
group_var <- quo(cyl)

ggplot(mtcars, aes(!!x_var, !!y_var)) + 
  geom_point() + 
  facet_wrap(vars(!!group_var))
```

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+p + facet_grid(vars(vs), vars(cyl), labeller = label_both)
+```
+
+### Nonstandard aesthetics
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Let's drop this here and save for the release post

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+```{r multiple-aes}
+ggplot(iris, aes(x = Species, y = Sepal.Length, color = Species, fill = Species)) +
+  geom_boxplot(alpha = 0.2) +
+  geom_point(position = "jitter") +
+  scale_colour_hue(c = 50, l = 40, aesthetics = c("colour", "fill"))
+```
+
+## And more...
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Can you please add brief sections for `geom_sf()`, `calc()` (soon to be `stat()`), and option for viridis colour scales.

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+A new release of ggplot2 (2.3.0) is on the horizon, and we'd love to have people 
+try it out, give us feedback, and [report issues](https://github.com/tidyverse/ggplot2/issues) 
+before we submit it to CRAN. This version is the culmination of over a year and 
+a half of development, not all of which will be captured here. So, please see [NEWS](https://github.com/tidyverse/ggplot2/blob/master/NEWS.md) for a fuller 
+account of updates and changes.
+
+In addition to highlighting a few features and improvements, we also want to 
+share a bit about our release-preparation process for ggplot2, which has over 
+1,700 reverse dependencies.
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Actually >2000 (including bioconductor)

File: content/articles/2018-05-ggplot2-2-3-0.Rmd [hunk 9]
```diff
@@ -0,0 +1,142 @@
+---
+title: ggplot2 2.3.0 — upcoming release
+author: Mara Averick
+date: '2018-05-18'
```
Comment (hadley) [inline content/articles/2018-05-ggplot2-2-3-0.Rmd]: Can you update for today please?

**Comment by batpigandme** [comment]
@hadley changes made, as requested.
